src/http_client.py

```diff
@@ -236,17 +238,18 @@ async def _get_user_json(self, message=None):
             if LOG_SENSITIVE_DATA:
                 log.debug(f"ROCKSTAR_OLD_AUTH: {old_auth}")
             else:
-                log.debug(f"ROCKSTAR_OLD_AUTH: {str(old_auth)[:5]}***{str(old_auth[-3:])}")
+                log.debug(f"ROCKSTAR_OLD_AUTH: ***")
             headers = {
-                "accept": "application/json, text/plain, */*",
-                "connection": "keep-alive",
-                "cookie": "ScAuthTokenData=" + old_auth,
-                "host": "www.rockstargames.com",
+                "accept": "*/*",
+                "cookie": await self.get_cookies_for_headers(),
                 "referer": "https://www.rockstargames.com",
                 "user-agent": USER_AGENT
             }
-            resp = await self._current_session.get(r"https://www.rockstargames.com/auth/get-user.json", headers=headers,
-                                                   allow_redirects=False)
+            resp = await self._current_session.get(r"https://www.rockstargames.com/graph.json?operationName=User&"
+                                                   r"variables=%7B%22locale%22%3A%22en_us%22%7D&extensions=%7B%22"
+                                                   r"persistedQuery%22%3A%7B%22version%22%3A1%2C%22sha256Hash%22%3A%22"
+                                                   r"6aa5127bff85d7fc23ffc192c74bb2e38c3c855482b33b2395aa103a554e9241"
+                                                   r"%22%7D%7D", headers=headers, allow_redirects=False)
```

Comment:
I'm a bit worried if this is reliable for a longer time. But for now let's try it out

Reply:
I am also, mainly because of the ``sha256Hash`` section of the URL. I have no idea where the hash is derived from, nor do I know if/when it might change. I will try looking further into that.

Reply:
So, after some debugging with the JS files on Rockstar's website, I found the string that generates the correct SHA-256 hash:

```
query User($code: String) {
  user(code: $code) {
    ...userFields
    __typename
  }
}

fragment userFields on RockstarGames_Users_Graph_Type_User {
  id
  nickname
  avatar
  profile_link
  dob
  __typename
}

```

(The string must include the intentional new-line character at the end.) Thus, it is apparent that Rockstar has adopted the use of GraphQL for their API, and the hash is derived from the request made for the query (see https://graphql.org/learn/queries/#fragments).

Reply:
cool. So are they sent a hash instead of putting the whole graphql query in url fragment?

Reply:
It seems like that is the case. Other requests to the graph use different hash values for different types of queries (i.e., for translations with https://www.rockstargames.com/graph.json?operationName=Translations&variables=%7B%22cache%22%3Atrue%2C%22config%22%3A%22default%22%2C%22locale%22%3A%22en_us%22%7D&extensions=%7B%22persistedQuery%22%3A%7B%22version%22%3A1%2C%22sha256Hash%22%3A%222a141a9b2fa947b1e4bcfe9f02ec800b8cf63bfae186902de329969d94f0e4fe%22%7D%7D).

The hash itself is generated in the JavaScript file located at https://www.rockstargames.com/dist/js/rockstar-site-7a21ed38f39767850392.js if you want to see it for yourself. (Since the file is minified, I would do a Ctrl-F for ``sha256Hash`` and then set a breakpoint at the function called there.)